### PR TITLE
Graphite: Fix drag and drop queries

### PR DESCRIPTION
**Pull request by ifrost**
**What this PR does / why we need it**:

In Dashboards: when an empty query (with "select metric" segment only) is dragged it's then an additional `onChange()` is incorrectly triggered (even though the query doesn't change). This leads to keeping stale state on the drop. See the GIF below how refId is set to B after the drop:

refid 2

A segment with value of "select segment" should be ignored but regular expression was working only if it's not the very first segment. I've updated the regex and added some tests.

**Which issue(s) this PR fixes**:

Fixes N/A

**Special notes for your reviewer**:

This will be needed for https://github.com/grafana/grafana/pull/38942 as well. CC: @Elfo404

### CODE REVIEW

File: public/app/plugins/datasource/graphite/specs/graphite_query.test.ts [hunk 1]
```diff
@@ -74,4 +74,25 @@ describe('Graphite query model', () => {
       expect(ctx.queryModel.functions[1].params[0]).toBe('$limit');
     });
   });
+
+  describe('skips "select metric" when target is rendered', () => {
```
Comment (Elfo404) [inline public/app/plugins/datasource/graphite/specs/graphite_query.test.ts:78]: can we find a better description for this test? it's not very clear to me who is skipping what, and what does "skips" mean in this context.

Comment (ifrost) [reply public/app/plugins/datasource/graphite/specs/graphite_query.test.ts]: better dfd1e571a483ff6f8d201e7065b0e80b66936960?

Comment (Elfo404) [reply public/app/plugins/datasource/graphite/specs/graphite_query.test.ts]: lovely <3

**Review by Elfo404 — APPROVED** [review]
🚀

**Pull request merged**